bitbox-bridge/release/Dockerfile

```diff
@@ -33,12 +33,12 @@ RUN gem install --no-document fpm
 ##############################
 # llvm 8 (needed because some rust crates compile C code)
 RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
-RUN add-apt-repository 'deb http://apt.llvm.org/xenial llvm-toolchain-xenial-8 main'
+RUN add-apt-repository 'deb http://apt.llvm.org/xenial llvm-toolchain-xenial-11 main'
 RUN apt-get update -y && apt-get install -y --no-install-recommends \
-    llvm-8 \
-    clang-8
-RUN update-alternatives --install /usr/bin/clang clang /usr/bin/clang-8 100
-RUN update-alternatives --install /usr/bin/clang++ clang++ /usr/bin/clang++-8 100
+    llvm-11 \
+    clang-11
```

Comment:
I couldn't get past this line: "Unable to locate package llvm-11" and clang-11.

![image](https://user-images.githubusercontent.com/25405/115521794-cdb0c500-a28b-11eb-9131-0826dcd1d7a9.png)

I'm actually surprised the whole thing always requires docker image and osxcross. I built the universal binary without docker and no issues, running these commands manually:

```shell
$ rustup default stable-x86_64-apple-darwin
$ cargo build --target x86_64-apple-darwin --release
$ file target/x86_64-apple-darwin/release/bitbox-bridge
target/x86_64-apple-darwin/release/bitbox-bridge: Mach-O 64-bit executable x86_64

$ rustup default stable-aarch64-apple-darwin
$ cargo build --target aarch64-apple-darwin --release
$ file target/aarch64-apple-darwin/release/bitbox-bridge
target/aarch64-apple-darwin/release/bitbox-bridge: Mach-O 64-bit executable arm64

$ lipo -create -output bitbox-bridge-uni bitbox-bridge-x86 bitbox-bridge-aarch64
$ file bitbox-bridge-uni 
bitbox-bridge-uni: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64]
bitbox-bridge-uni (for architecture x86_64):    Mach-O 64-bit executable x86_64
bitbox-bridge-uni (for architecture arm64):    Mach-O 64-bit executable arm64
```

If I don't `rustup default stable-x86_64-apple-darwin` and just do `cargo build --target x86_64-apple-darwin --release` while the default is aarch64, I get this eror:

```
$ cargo build --target x86_64-apple-darwin --release
   Compiling bitbox-bridge v1.3.0 (/Users/dev/code/bitbox-bridge/bitbox-bridge)
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-m64" "-arch" "x86_64" "-L" "/Users/dev/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/Users/dev/code/bitbox-bridge/target/x86_64-apple-darwin/release/deps/bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o" "-o" "/Users/dev/code/bitbox-bridge/target/x86_64-apple-darwin/release/deps/bitbox_bridge" "-Wl,-dead_strip" "-nodefaultlibs" "-L" "/Users/dev/code/bitbox-bridge/target/x86_64-apple-darwin/release/deps" "-L" "/Users/dev/code/bitbox-bridge/target/release/deps" "-L" "/Users/dev/code/bitbox-bridge/target/x86_64-apple-darwin/release/build/hidapi-8b1a006045e6977b/out" "-L" "/Users/dev/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib" "/var/folders/vy/k8bwyxjn3gl82q9xnflthzxc0000gp/T/rustcE9bEMc/libhidapi-f5e0d83556ec0aba.rlib" "/Users/dev/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/x86_64-apple-darwin/lib/libcompiler_builtins-10db70d883838cbc.rlib" "-framework" "Security" "-framework" "IOKit" "-framework" "CoreFoundation" "-lSystem" "-lresolv" "-lc" "-lm"
  = note: ld: warning: ignoring file /var/folders/vy/k8bwyxjn3gl82q9xnflthzxc0000gp/T/rustcE9bEMc/libhidapi-f5e0d83556ec0aba.rlib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
          Undefined symbols for architecture x86_64:
            "_hid_enumerate", referenced from:
                hidapi::HidApi::get_hid_device_info_vector::h8b637a262beebcc1 in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
            "_hid_free_enumeration", referenced from:
                hidapi::HidApi::get_hid_device_info_vector::h8b637a262beebcc1 in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
            "_hid_error", referenced from:
                hidapi::HidDevice::check_error::h4cfdf7834a51e571 in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
            "_hid_close", referenced from:
                alloc::sync::Arc$LT$T$GT$::drop_slow::h137692c626eb9049 in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
                core::ptr::drop_in_place$LT$hidapi..HidDevice$GT$::h240b36821f117494 (.11655) in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
            "_hid_read", referenced from:
                std::sys_common::backtrace::__rust_begin_short_backtrace::hedf06b6242c8bdad in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
            "_hid_write", referenced from:
                _$LT$$RF$mut$u20$T$u20$as$u20$futures_io..if_std..AsyncWrite$GT$::poll_write::h74e007385314096e in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
            "_hid_init", referenced from:
                bitbox_bridge::main::hfb7940008c45e7c9 in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
            "_hid_exit", referenced from:
                bitbox_bridge::main::hfb7940008c45e7c9 in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
                alloc::sync::Arc$LT$T$GT$::drop_slow::h793e7207a23e0f23 in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
                core::ptr::drop_in_place$LT$hidapi..HidApiLock$GT$::ha3f65a2732b78843 in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
            "_hid_open_path", referenced from:
                _$LT$core..future..from_generator..GenFuture$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h247d2b4eafd7e85f in bitbox_bridge.bitbox_bridge.6ojjinjr-cgu.0.rcgu.o
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I suspect somewhere in the middle aarch64 is used (bug?) and hid lib is built for wrong architecture. But this is a minor issue imho. Maybe setting some additional rustc flags would fix it, but setting the default arch before building the whole thing also works.

Oh and I also ` mv .cargo/config .cargo/config.org` before starting to compile anything, just in case.

Can we get rid of osxcross?

Reply:
This build script is for Linux, so we need it, one option is to have build instructions for osx as well which should be simpler